src/mon/OSDMonitor.cc

```diff
@@ -6987,6 +6987,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
   bool ret = false;
   stringstream ss;
   string rs;
+  std::string sure;
```

Comment:
I'm not convinced this is a good idea, it looks like a 'C' optimisation to me. In C++ you declare as late as possible and minimise the lifetime of an object. Are we confident that if we call cmd_getval() once and "sure" gets set to "--yes-i-really-mean-it" we don't subsequently test for "--yes-i-really-mean-it" without clearing it? If we call cmd_getval() and the value of "sure" is already "--yes-i-really-mean-it" it will remain set even if the call fails. In short I think using a more global "sure" variable here makes the code more complex to reason about and maintain and I prefer the duplicate local versions.

Reply:
@badone 
Thanks for your input.
I completely forgot about RAII, will update then ping you to get your suggestion.